Support Elasticsearch monitoring (#10690)
apache/skywalking: docs/en/setup/backend/backend-elasticsearch-monitoring.md

# Elasticsearch monitoring

SkyWalking leverages elasticsearch-exporter for collecting metrics data from Elasticsearch. It leverages OpenTelemetry
Collector to transfer the metrics to
[OpenTelemetry receiver](opentelemetry-receiver.md) and into the [Meter System](./../../concepts-and-designs/meter.md).

## Data flow

1. The elasticsearch-exporter collect metrics data from Elasticsearch.
2. OpenTelemetry Collector fetches metrics from elasticsearch-exporter via Prometheus Receiver and pushes metrics to
   SkyWalking OAP Server via OpenTelemetry gRPC exporter.
3. The SkyWalking OAP Server parses the expression with [MAL](../../concepts-and-designs/mal.md) to
   filter/calculate/aggregate and store the results.

## Setup

1. Setup [elasticsearch-exporter](https://github.com/prometheus-community/elasticsearch_exporter).

> 2. Set up [OpenTelemetry Collector ](https://opentelemetry.io/docs/collector/getting-started/#kubernetes). For details
>    on Redis Receiver in OpenTelemetry Collector, refer

Reviewer: ```suggestion
2. Set up [OpenTelemetry Collector ](https://opentelemetry.io/docs/collector/getting-started/#kubernetes). The example for OpenTelemetry Collector configuration, refer
```

> # Elasticsearch monitoring

Reviewer 2: You need to update `menu.yml` to add the ref to this file.

SkyWalking leverages elasticsearch-exporter for collecting metrics data from Elasticsearch. It leverages OpenTelemetry
Collector to transfer the metrics to
[OpenTelemetry receiver](opentelemetry-receiver.md) and into the [Meter System](./../../concepts-and-designs/meter.md).

## Data flow

1. The elasticsearch-exporter collect metrics data from Elasticsearch.
2. OpenTelemetry Collector fetches metrics from elasticsearch-exporter via Prometheus Receiver and pushes metrics to
   SkyWalking OAP Server via OpenTelemetry gRPC exporter.
3. The SkyWalking OAP Server parses the expression with [MAL](../../concepts-and-designs/mal.md) to
   filter/calculate/aggregate and store the results.

## Setup

1. Setup [elasticsearch-exporter](https://github.com/prometheus-community/elasticsearch_exporter).
2. Set up [OpenTelemetry Collector ](https://opentelemetry.io/docs/collector/getting-started/#kubernetes). For details
   on Redis Receiver in OpenTelemetry Collector, refer
   to [here](../../../../test/e2e-v2/cases/elasticsearch/otel-collector-config.yaml).
3. Config SkyWalking [OpenTelemetry receiver](opentelemetry-receiver.md).

## Elasticsearch Monitoring

> Elasticsearch monitoring provides multidimensional metrics monitoring, such as cluster, node, and index. 
> 
> Elasticsearch cluster as a `Service` in OAP, Elasticsearch node as an `Instance` in OAP, Elasticsearch Index as an `Endpoint` in OAP, and land on the `Layer: Elasticsearch`.

Reviewer 2: ```suggestion
Elasticsearch monitoring provides multidimensional metrics monitoring of Elasticsearch clusters as `Layer: ELASTICSEARCH` `Service` in the OAP. In each cluster, the nodes are represented as `Instance` and indices are `Endpoint`s.
```
